overlays/musl.nix

```diff
@@ -0,0 +1,9 @@
+self: super: super.lib.optionalAttrs super.stdenv.hostPlatform.isMusl {
+  # musl appears to need a static openssl to build
+  openssl = (super.openssl.override { static = true; })
```

Comment:
I still think we're mixing up two things:
- The configure script thing, which is just plain wrong platform detection on musl, and could really be included unconditionally in an overlay guarded by the hostPlatform check.
- The changes to make the static build work, which we may only need for our fully-static test.

Reply:
Yes. The linux-generic64 problem seem to be independent of static vs shared.  Also it turns out the no_shared flag is not necessary (at least for the fully-static test).